rasa/core/test.py

```diff
@@ -373,7 +374,10 @@ def inline_comment(self) -> Optional[Text]:
         """A comment attached to this event. Used during dumping."""
         from rasa.shared.core.events import format_message
 
-        if self.predicted_intent != self.intent["name"]:
+        if (
+            self.predicted_intent != self.intent["name"]
+            or self.predicted_entities != self.target_entities
```

Comment:
I'm wondering if this shouldn't go into an `elif` and using `inline_comment_for_entity` in that elif branch? This solution causes a different test to fail with correct intent but wrong entity and it's perhaps clearer if the comment points to the entity as the source of failure rather than intent.
Probably off-topic, I also found it strange that this static method is not used anywhere 🤔

Reply:
`inline_comment_for_entity` is actually used by `YAMLStoryWriter` (in `process_user_utterance` method).
Regarding the first suggestion: not sure how it can help with that test but still thinking of it :(

Reply:
Found another way of doing it. Please, re-check!